lib/process-services-cloud/src/lib/form/components/widgets/data-table/helpers/data-table-path-parser.helper.ts

```diff
@@ -17,21 +17,38 @@
 
 export class DataTablePathParserHelper {
     private readonly removeSquareBracketsRegEx = /^\[(.*)\]$/;
+    private readonly indexReferencesRegEx = /(\[\d+\])+$/;
 
     retrieveDataFromPath(data: any, path: string): any[] {
         const properties = this.splitPathIntoProperties(path);
-        const currentProperty = this.removeSquareBracketsFromProperty(properties.shift());
+        let currentProperty = properties.shift();
+        const propertyIndexReferences = this.getIndexReferencesFromProperty(currentProperty);
+
+        const isPropertyWithSingleIndexReference = propertyIndexReferences.length === 1;
+        const isPropertyWithMultipleIndexReferences = propertyIndexReferences.length > 1;
+
+        if (isPropertyWithMultipleIndexReferences) {
+            return [];
+        }
+
+        currentProperty = isPropertyWithSingleIndexReference
+            ? this.removeSquareBracketsAndIndexReferencesFromProperty(currentProperty)
+            : this.removeSquareBracketsFromProperty(currentProperty);
 
         if (!this.isPropertyExistsInData(data, currentProperty)) {
             return [];
         }
 
-        const nestedData = data[currentProperty];
+        const nestedData = isPropertyWithSingleIndexReference ? data[currentProperty][propertyIndexReferences[0]] : data[currentProperty];
 
         if (Array.isArray(nestedData)) {
             return nestedData;
         }
 
+        if (properties.length === 0) {
+            return [];
```

Comment:
the failing cases should be at the top not in the middle

Reply:
good point, thx